### PR TITLE
[SPARK-42138][UI] Handle null string values in JobData/TaskDataWrapper/ExecutorStageSummaryWrapper

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

Similar to https://github.com/apache/spark/pull/39666, this PR handles null string values in JobData/TaskDataWrapper/ExecutorStageSummaryWrapper
### Why are the changes needed?

Properly handles null string values in the protobuf serializer.
### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

New UTs

### CODE REVIEW

**Comment by gengliangwang** [comment]
cc @LuciferYang @panbingkun @techaddict

**Review by dongjoon-hyun — APPROVED** [review]
+1, Looks consistent (Pending CIs).

**Review by LuciferYang — APPROVED** [review]
+1, LGTM

**Comment by LuciferYang** [comment]
@gengliangwang @dongjoon-hyun I add some new tasks and will start work later, @panbingkun @techaddict feel free to take anyone from the them and leave a comment that you are working on it:

- SPARK-42139: Handle null string values in SQLExecutionUIData/SQLPlanMetric
- SPARK-42140: Handle null string values in ApplicationEnvironmentInfo/RuntimeInfo/PairStrings/ExecutorResourceRequest/TaskResourceRequest
- SPARK-42141: Handle null string values in ApplicationInfo/ApplicationAttemptInfo
- SPARK-42142: Handle null string values in CachedQuantile/ExecutorSummary/PoolData
- SPARK-42143: Handle null string values in RDDStorageInfo/RDDDataDistribution/RDDPartitionInfo
- SPARK-42144: Handle null string values in StageData/StreamBlockData/StreamingQueryData
- SPARK-42145: Handle null string values in SparkPlanGraphNode/SparkPlanGraphClusterWrapper

Some may be further consolidated into one task

And I re-open [SPARK-41677](https://issues.apache.org/jira/browse/SPARK-41677)

File: core/src/main/protobuf/org/apache/spark/status/protobuf/store_types.proto [hunk 1]
```diff
@@ -83,10 +83,10 @@ message TaskDataWrapper {
   int64 launch_time = 5;
   int64 result_fetch_start = 6;
   int64 duration = 7;
-  string executor_id = 8;
-  string host = 9;
-  string status = 10;
-  string task_locality = 11;
+  optional string executor_id = 8;
```
Comment (LuciferYang) [inline core/src/main/protobuf/org/apache/spark/status/protobuf/store_types.proto:86]: Have we considered defining a message as follows? 

```
message NullableString {
  bool is_null = 1;
  string value = 2;
}
```
In this way, we can distinguish the string that are originally optional, but we need to record one more state

Comment (gengliangwang) [reply core/src/main/protobuf/org/apache/spark/status/protobuf/store_types.proto:86]: Yeah that's complicated

**Comment by gengliangwang** [comment]
@dongjoon-hyun @LuciferYang Thanks for the review. Merging to master

**Pull request closed without merging**